include/real/real_data.hpp

```diff
@@ -183,9 +183,11 @@ namespace boost {
                     bool deviation_upper_boundary, deviation_lower_boundary;
 
                     /* if the interval contains zero, iterate until it doesn't, or until maximum_precision. */
-                    while (!ro.get_rhs_itr().get_interval().positive() &&
-                           !ro.get_rhs_itr().get_interval().negative() &&
-                           _precision <= this->maximum_precision())
+                   while (((!ro.get_rhs_itr().get_interval().positive() 
+                            && !ro.get_rhs_itr().get_interval().negative() ) 
+                            || ro.get_rhs_itr().get_interval().lower_bound == literals::zero_exact<T>
+                            || ro.get_rhs_itr().get_interval().upper_bound == literals::zero_exact<T> ) 
```

Comment:
Before checking if lower or upper bounds are equal to zero you should check if both boundaries are equals or not to each other, becose if boundaires are equals, then you already now the number it actually is zero and the divide by zero error should be thrown in that case. Now you are throwing `divergent_division_result_exception()` for both cases, but in the case of zero you should throw `division_by_zero()`

Reply:
No. It is not showing `divergent_division_result_exception()` error. The system will show `division_by_zero()`. 
I can put an additional check inside the loop, which will check equality of upper and lower bound of input and will throw `division_by_zero()` error. But, that seems unnecessary, if we get to a point when both lower and upper bound are equal and zero, then it is clear that we already go to maximum precision and exit the loop. Then, if the number is zero, then the required error will be thrown by the division system. That check inside the loop will be redundant.

Reply:
condition `(!ro.get_rhs_itr().get_interval().positive() && !ro.get_rhs_itr().get_interval().negative() )` evaluates to false for equal boundaries.

Reply:
Yes, and this is the condition for `divergent_divsion_exception()`, so the system will not throw that error for division by zero case.